app/commands/mentor/request/lock.rb

```diff
@@ -5,17 +5,14 @@ class Lock
 
       initialize_with :request, :mentor
 
-      # TODO: (Required) Add a guard that a mentor should only have
-      # two solutions locked to them at any time. When someone
-      # hits 'start discussion' that should remove the lock.
-
       def call
         ActiveRecord::Base.transaction do
           # This is a DB lock (The naming is confusing!)
           request.lock!
 
           # Guard against not being lockable
           raise SolutionLockedByAnotherMentorError unless request.lockable_by?(mentor)
+          raise MentorSolutionLockLimitReachedError if mentor.locks.size >= Mentor::RequestLock::MAX_LOCKS_PER_MENTOR
```

Comment:
Is `size` the right method to call here?

Reply:
Yeah, it'll proxy to `count` as locks hasn't been loaded as an array. So you could explicitly use `count` but I think `size` is fine too.